clang/lib/Driver/Driver.cpp

```diff
@@ -1846,12 +1846,18 @@ void Driver::PrintHelp(bool ShowHidden) const {
 }
 
 llvm::Triple Driver::MakeSYCLDeviceTriple(StringRef TargetArch) const {
-  llvm::Triple TT;
-  TT.setArchName(TargetArch);
-  TT.setVendor(llvm::Triple::UnknownVendor);
-  TT.setOS(llvm::Triple::UnknownOS);
-  TT.setEnvironment(llvm::Triple::SYCLDevice);
-  return TT;
+  SmallVector<StringRef, 5> SYCLAlias = {"spir", "spir64", "spir64_fpga",
+                                         "spir64_x86_64", "spir64_gen"};
+  if (std::find(SYCLAlias.begin(), SYCLAlias.end(), TargetArch) !=
+      SYCLAlias.end()) {
+    llvm::Triple TT;
+    TT.setArchName(TargetArch);
```

Comment:
As an afterthought, would something like this work too?
```suggestion
  llvm::Triple TT;
  TT.setArchName(TargetArch);
  if TT.isSPIR() {
```

Reply:
Doing something like this will override user settings if they want something different; say `spir64-intel-unknown` would be converted to `spir64-unknown-unknown-sycldevice`.  Being specific on what we see from the user and only setting if the explicit alias is used seems to contain the behavior better.

Reply:
Fair enough; we could however extend the logic in a way that would only enforce "unknown" in case the triple component value had not been specified (this would maybe allow for `spir64-intel` to be used on CLI as well). Anyway, I agree that the current approach is best for the initial support.